File: erp-web/src/api/interface/goods/product.ts
/**
 * 本文件由 pnpm gen:page 生成(spec 行式拍板 + tools/openapi.json 快照)
 * 默认存在即跳过:人工改动不会被 --force 之外的任何方式覆盖;重新生成前先 diff 人工改动
 * 框架代码禁手改;业务槽位一律 TODO(编号),编号已登记 TODO.md
 */

/** 商品管理接口类型(gen:page 从 openapi 快照生成;对齐后端契约,禁手抄字段) */

/** 商品管理实体(分页行 / 详情) */
export interface ProductResponse {
  /** id */
  id: number
  /** SPU编码 */
  spuCode: string
  /** 商品名称 */
  name: string
  /** 分类ID */
  categoryId: number
  /** 品牌ID */
  brandId: number
  /** attrsJson */
  attrsJson: string
  /** 状态 */
  status: number
  /** 创建时间 */
  createdAt: string
  /** updatedAt */
  updatedAt: string
}

/** 商品管理分页查询入参(不含分页参数) */
export interface ProductQuery {
  /** 关键字 */
  keyword?: string
  /** 分类ID */
  categoryId?: number
}

File: erp-goods/src/main/java/com/own/erp/goods/entity/Product.java
package com.own.erp.goods.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : chenyi
 * @Date : 2026/9/3
 * @Description : 商品 SPU(product):内部商品库主档
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("product")
public class Product {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 内部 SPU 编码,唯一 */
    private String spuCode;

    /** 商品名称 */
    private String name;

    /** 分类ID(product_category.id) */
    private Long categoryId;

    /** 品牌ID(brand.id) */
    private Long brandId;

    /** 销售属性(JSON),如 [{"name":"颜色","values":["黑","白"]}] */
    private String attrsJson;

    /** 1=在售 0=停用 */
    private Integer status;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
